client/web/src/search/SearchConsolePage.tsx

```diff
@@ -46,11 +46,10 @@ export const SearchConsolePage: React.FunctionComponent<React.PropsWithChildren<
     const { globbing, streamSearch, extensionsController, isSourcegraphDotCom } = props
     const extensionHostAPI =
         extensionsController !== null && window.context.enableLegacyExtensions ? extensionsController.extHostAPI : null
-    const enableGoImportsSearchQueryTransform = useExperimentalFeatures(
-        features => features.enableGoImportsSearchQueryTransform
-    )
-    const applySuggestionsOnEnter =
-        useExperimentalFeatures(features => features.applySearchQuerySuggestionOnEnter) ?? true
+    const { enableGoImportsSearchQueryTransform, applySuggestionsOnEnter } = useExperimentalFeatures(features => ({
+        enableGoImportsSearchQueryTransform: features.enableGoImportsSearchQueryTransform,
+        applySuggestionsOnEnter: features.applySearchQuerySuggestionOnEnter ?? true,
+    }))
```

Comment:
It looks like we don't need a selector function in most cases. Would it be good enough to return the experimental features object by default to avoid redundant field mapping?

```suggestion
    const { enableGoImportsSearchQueryTransform, applySuggestionsOnEnter = true } = useExperimentalFeatures()
```

1. As we discussed, there should be no performance issues caused by this update because the experimental features object is not mutable. We load it once and use it during the lifetime of the application. Updates are rare, and it's ok to re-rerender UI when they happen.
2. It's important to discourage the renaming of things. With the selector API, it's easy to save keystrokes and name these settings as something else. Then find-and-replace doesn't work as well as it should, and it makes it harder to follow the logic in components.

Reply:
The object is mutated when the user settings change and in at least one feature that I’m aware of it is also mutated from inside the app code. I don't think the mental model of this being static is safe to apply here, but yeah it's rarely changed which makes it great for context.

My thinking was that we keep the API so we don't have to update all call sites and we can in the future use https://github.com/reactjs/rfcs/pull/119 to make things faster again.

I can look into the change if you feel strongly though!

Reply:
Sounds good, let's keep it and also save time on refactoring 👍